### PR TITLE
feat(type-on-hover): Add type-on-hover on boolean and natural literals

### PR DESCRIPTION
**Pull request by german1608**
This is the first implementation of the type-on-hover feature. It is currently a draft and it is intended to be really small (excluding the golden file updates).

The reason for this PR to be that small is because this idea will be expanded on further PRs related to type-on-hover. So please let me know what you think and what can be improved before working on the next elements that will have this feature.

The tooltip library that I'm using is https://atomiks.github.io/tippyjs/, which is really lightweight.

I wanted to add jump-to-definition on imports before this, but I got blocked on that... didn't found the best way to do it. My problem is with implementation, I got an idea which I described [here (comment)](https://github.com/dhall-lang/dhall-haskell/issues/1994#issuecomment-675622903). Path manipulation was really messy and I felt that working on this feature first might be better because I already had a clearer idea about how to do this and I prefer to deliver as much work before ending the program as I can.

To end, I feel that the `dhall-docs` package structure is not the best... do you have any advice about organizing Haskell packages? I feel that the entire tool can fit roughly in a single file :laughing:

### CODE REVIEW

**Comment by german1608** [comment]
Please check out this whenever you have time and let me know what you think @Gabriel439 @sjakobi @Profpatsch

**Comment by german1608** [comment]
https://hydra.dhall-lang.org/build/70598/download/1/docs/TypeOnHover.dhall.html Checkout this simple example

File: dhall-docs/src/Dhall/Docs/CodeRenderer.hs [hunk 1]
```diff
@@ -239,6 +239,28 @@ fragments = Data.List.sortBy sorter . removeUnusedDecls . Writer.execWriter . in
...
+          where
+            _type = TypeFromSource $ case TypeCheck.typeOf (BoolLit b) of
+                Left _ -> fileAnIssue "typeOfStatic"
+                Right e -> e
+
+        Note src (NaturalLit n) -> Writer.tell [SourceCodeFragment src $ StaticExpression _type] >> return NoInfo
+          where
+            _type = TypeFromSource $ case TypeCheck.typeOf (NaturalLit n) of
+                Left _ -> fileAnIssue "typeOfStatic"
+                Right e -> e
```
Comment (Gabriel439) [inline dhall-docs/src/Dhall/Docs/CodeRenderer.hs:262]: I recommend studying how `dhall-lsp-server` does this:

https://github.com/dhall-lang/dhall-haskell/blob/master/dhall-lsp-server/src/Dhall/LSP/Backend/Typing.hs

For example, one trick you can reuse from that code is to use the `subExpressions` `Traversal` to detect if a term has children or not when deciding whether or not to create a type tooltip

**Comment by Gabriel439** [comment]
I'll close this for now since the work has stalled

**Pull request closed without merging**